### PR TITLE
use trailing_zeros for threadset iteration

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- Checking many indexes that do not need to be checked
- Have seen `select_thread` showing up in profiles, think it is likely due to this

#### Summary of Changes
- Use `trailing_zeros` and modify value to iterator thread-sets

Fixes #

### CODE REVIEW

File: core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs [hunk 1]
```diff
@@ -430,6 +430,22 @@ impl ThreadSet {
...
+
+impl Iterator for ThreadSetIterator {
+    type Item = ThreadId;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.0 == 0 {
+            None
+        } else {
+            let thread_id = self.0.trailing_zeros() as ThreadId;
+            self.0 &= self.0 - 1;
```
Comment (bw-solana) [inline core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs:443]: I had to stare at this for longer than I'd care to admit, but looks good 👍

**Review by bw-solana — APPROVED** [review]
LGTM

File: core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs (hunk 1 above)
Comment (ksolana) [reply core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs:443]: neat trick!

Comment (t-nelson) [reply core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs:443]: > I had to stare at this for longer than I'd care to admit, but looks good 👍

this is usually sufficient to justify a code comment for the sake of future maintainers

**Pull request merged**